src/main/java/com/powsybl/network/hypothesis/GeneratorsStartupAlgorithm.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.hypothesis;
+
+import com.powsybl.iidm.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * @author Chamseddine BENHAMED <chamseddine.benhamed at rte-france.com>
+ */
+public class GeneratorsStartupAlgorithm {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeneratorsStartupAlgorithm.class);
+
+    List<Generator> startupGroupsPowerMax = new ArrayList<>();
+
+    double nuclearBandSetting = 0.05;
+    double thermalBandSetting = 0.05;
+    double hydroBandSetting = 0.1;
+
+    double lossCoefficient = 0;
+    double defaultAbatementCoefficient = 0;
+
+    StartupMarginalGroupType startUpMarginalGroupType = StartupMarginalGroupType.CLASSIC;
+
+    //thresholds for voltage adjustment
+    double pThreshold = 0.0;
+    double qThreshold = 0.0;
+
+    /**
+     * starts generators groups using Classic or Mexico algorithm
+     * @param network
+     * @param startUpMarginalGroupType Mexico or Classic
+     * @param defaultAbatementCoefficient defaultAbatementCoefficient parameter
+     * @param lossCoefficient lossCoefficient parameter
+     * @param pThreshold pThreshold parameter
+     * @param qThreshold qThreshold parameter
+     * @param startupGroupsPowerMax list of groups that should be started at Pmax if picked
+     */
+    public void apply(Network network, StartupMarginalGroupType startUpMarginalGroupType, double defaultAbatementCoefficient,
+                      double lossCoefficient, double pThreshold, double qThreshold, List<Generator> startupGroupsPowerMax) {
+        //parameters initialization
+        this.lossCoefficient = lossCoefficient;
+        this.defaultAbatementCoefficient = defaultAbatementCoefficient;
+        this.pThreshold = pThreshold;
+        this.qThreshold = qThreshold;
+        this.startUpMarginalGroupType = startUpMarginalGroupType;
+        this.startupGroupsPowerMax = startupGroupsPowerMax;
+
+        // prepare startupGroupsPerConnectedComponent HashMap
+        Map<Component, List<StartupGroup>> startupGroupsPerConnectedComponent = prepareStartupGroupsPerConnectedComponentMap(network);
+
+        // for each connected component
+        startupGroupsPerConnectedComponent.keySet().forEach(component -> {
```

Comment:
Since you use key and value, you don't have to use `keySet()`, you can directly use `forEach`